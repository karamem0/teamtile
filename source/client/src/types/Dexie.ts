//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export interface ArrayEntity<T> {
  id: string,
  expired: number,
  values: T[]
}

export interface ValueEntity<T> {
  id: string,
  expired: number,
  value: T
}
