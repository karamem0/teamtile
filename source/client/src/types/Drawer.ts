//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export type DrawerType =
  | 'channel'
  | 'member'
  | 'tag';
