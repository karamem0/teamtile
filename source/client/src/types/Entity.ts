//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export enum MembershipType {
  private = 'private',
  standard = 'standard'
}

export enum VisibilityType {
  private = 'private',
  public = 'public'
}

export enum AccordionType {
  all = 'all',
  archived = 'archived',
  pinned = 'pinned'
}

export interface Channel {
  id?: string,
  displayName?: string,
  membershipType?: MembershipType,
  webUrl?: string
}

export interface Drive {
  id?: string,
  webUrl?: string
}

export interface Group {
  id?: string,
  email?: string,
  sensitivityLabel?: string
}

export interface Icon {
  id?: string,
  data?: string
}

export interface Member {
  id?: string,
  displayName?: string,
  email?: string,
  icon?: string,
  userId?: string
}

export interface Tab {
  id?: string,
  appId?: string,
  displayName?: string,
  webUrl?: string
}

export interface Team {
  id?: string,
  archived?: boolean,
  description?: string,
  displayName?: string,
  icon?: string,
  internalId?: string,
  visibility?: VisibilityType,
  webUrl?: string
}

export interface TeamInfo {
  id?: string,
  displayName?: string,
  tenantId?: string
}
