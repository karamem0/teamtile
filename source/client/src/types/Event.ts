//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

export type Event = React.SyntheticEvent | React.SyntheticEvent<unknown> | Record<string, never>;

export type EventHandler<T = never> = (
  event?: Event,
  data?: T
) => void;
