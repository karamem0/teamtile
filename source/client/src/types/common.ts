//
// Copyright (c) 2021 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/master/LICENSE
//

export interface KeyValue<K, V> {
  key: K,
  value: V
}
