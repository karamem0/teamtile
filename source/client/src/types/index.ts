//
// Copyright (c) 2021 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/master/LICENSE
//

export interface Team {
  id?: string;
  internalId?: string;
  name?: string;
  description?: string;
  visibility?: string;
  url?: string;
  icon?: string;
}

export interface TeamChannels {
  count?: number;
  nextLink?: string;
  values?: TeamChannel[];
}

export interface TeamChannel {
  id?: string;
  name?: string;
  url?: string;
}

export interface TeamMembers {
  count?: number;
  nextLink?: string;
  values?: TeamMember[];
}

export interface TeamMember {
  id?: string;
  name?: string;
  email?: string;
}

export interface TeamDrive {
  id?: string;
  url?: string;
}
