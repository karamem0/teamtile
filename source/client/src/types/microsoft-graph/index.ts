import * as microsoftGraph from '@microsoft/microsoft-graph-types';

export type MicrosoftGraphResponse = MicrosoftGraphValueResponse | MicrosoftGraphArrayResponse | MicrosoftGraphErrorResponse | string;

export interface MicrosoftGraphValueResponse extends microsoftGraph.Entity
{
  '@odata.context': string;
}

export interface MicrosoftGraphArrayResponse {
  '@odata.context': string;
  '@odata.count': number;
  '@odata.nextLink': string;
  value: microsoftGraph.Entity[];
}

export interface MicrosoftGraphBatchResponse {
  id: string;
  status: number;
  body: MicrosoftGraphResponse;
  headers: {
    [key: string]: string;
  };
}

export interface MicrosoftGraphErrorResponse {
  error: microsoftGraph.GenericError;
}
