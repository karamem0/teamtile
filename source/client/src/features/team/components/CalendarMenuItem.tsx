//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';
import { useAsyncFn, useError } from 'react-use';

import {
  app,
  appInstallDialog
} from '@microsoft/teams-js';

import { useSnackbar } from '../../../providers/SnackbarProvider';
import { SnackbarType } from '../../../types/Snackbar';
import { Item } from '../../../types/Store';
import { isPC } from '../../../utils/Teams';
import { getTab } from '../managers/TeamManager';

import Presenter from './CalendarMenuItem.presenter';

interface CalendarMenuItemProps {
  item?: Item
}

function CalendarMenuItem(props: CalendarMenuItemProps) {

  const { item } = props;

  const { setSnackbar } = useSnackbar();
  const dispatchError = useError();
  const [ state, fetch ] = useAsyncFn((teamId: string, channelId: string, appId: string) => getTab(teamId, channelId, appId));

  React.useEffect(() => {
    if (!state.error) {
      return;
    }
    dispatchError(state.error);
  }, [
    dispatchError,
    state.error
  ]);

  const handleClick = React.useCallback(async () => {
    if (!item?.value?.id) {
      return;
    }
    if (!item?.value?.internalId) {
      return;
    }
    if (isPC(await app.getContext())) {
      const appId = process.env.VITE_CALENDAR_APP_ID;
      const appTab = await fetch(item.value.id, item.value.internalId, appId);
      if (appTab?.webUrl) {
        return await app.openLink(appTab.webUrl);
      } else if (appInstallDialog.isSupported()) {
        return await appInstallDialog.openAppInstallDialog({ appId });
      }
    }
    setSnackbar({
      text: 'This operation is not supported on this device.',
      type: SnackbarType.warning
    });
  }, [
    fetch,
    item,
    setSnackbar
  ]);

  return (
    <Presenter
      loading={state.loading}
      onClick={handleClick} />
  );

}

export default CalendarMenuItem;
