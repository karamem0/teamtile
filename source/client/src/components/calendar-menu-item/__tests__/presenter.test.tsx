//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import { Group } from '../../../types/entity';
import CalendarMenuItem from '../presenter';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('CalendarMenuItem', () => {

  it('create shapshot', async () => {
    const json = await import('./__jsons__/presenter.test.json');
    const params = {
      group: json.default as Group,
      onClick: jest.fn()
    };
    render(<CalendarMenuItem {...params} />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
