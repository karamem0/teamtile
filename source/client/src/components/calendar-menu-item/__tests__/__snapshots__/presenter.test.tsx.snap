// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarMenuItem create shapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-1 {
  line-height: 1em;
}

.emotion-1 span {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<body>
  <div>
    <span
      class="ui-text emotion-0"
      data-uses-unhanded-props="true"
      role="button"
    >
      <span
        class="emotion-1"
      >
        <span
          aria-hidden="true"
          class="root_dd790ee3"
        >
          <svg
            class="svg_dd790ee3"
            focusable="false"
            viewBox="0 0 2048 2048"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M768 768h128v128H768V768zm384 768h128v128h-128v-128zm384-768h128v128h-128V768zm-384 0h128v128h-128V768zm-384 256h128v128H768v-128zm-384 0h128v128H384v-128zm1152 0h128v128h-128v-128zm-384 0h128v128h-128v-128zm-384 256h128v128H768v-128zm-384 0h128v128H384v-128zm1152 0h128v128h-128v-128zm-384 0h128v128h-128v-128zm-384 256h128v128H768v-128zm-384 0h128v128H384v-128zM2048 128v1792H0V128h384V0h128v128h1024V0h128v128h384zM128 256v256h1792V256h-256v128h-128V256H512v128H384V256H128zm1792 1536V640H128v1152h1792z"
            />
          </svg>
        </span>
      </span>
    </span>
  </div>
</body>
`;
